File: app/src/main/java/com/example/user/biblejournal/view/MainActivity.java
package com.example.user.biblejournal.view;

import android.os.Bundle;

import androidx.appcompat.app.AppCompatActivity;

import com.example.user.biblejournal.R;
import com.example.user.biblejournal.view.editnote.EditNoteFragment;
import com.example.user.biblejournal.view.notelist.NoteListFragment;

public class MainActivity extends AppCompatActivity implements EditNoteFragment.EditNoteListener, NoteListFragment.NoteListListener {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        if (null == savedInstanceState) {
            getSupportFragmentManager().beginTransaction()
                    .add(R.id.fragment_placeholder, NoteListFragment.newInstance())
                    .commit();
        }
    }

    @Override
    public void backPreviousScreen() {
        // TODO: Use Jetpack Navigation component
        getSupportFragmentManager().popBackStack();
    }

    @Override
    public void startEditNote(Integer noteId) {
        getSupportFragmentManager()
                .beginTransaction()
                .replace(R.id.fragment_placeholder, EditNoteFragment.newInstance(noteId))
                .addToBackStack(null)
                .commit();
    }


}

File: app/src/main/java/com/example/user/biblejournal/main/view/MainActivity.kt
package com.example.user.biblejournal.main.view

import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import androidx.fragment.app.Fragment
import androidx.navigation.NavController
import androidx.navigation.Navigation
import com.example.user.biblejournal.R
import com.example.user.biblejournal.writer.view.WriterFragment

class MainActivity : AppCompatActivity(), WriterFragment.EditNoteListener {
    override fun onBackPressed() {
        val navHostFragment = supportFragmentManager.findFragmentById(R.id.main_nav_host_fragment)
        val fragmentWithBackPressedListener = navHostFragment!!.childFragmentManager.fragments.stream().filter { f: Fragment? -> f is MainActivityBackPressListener }.findFirst()
        if (fragmentWithBackPressedListener.isPresent) {
            val listener: MainActivityBackPressListener = fragmentWithBackPressedListener.get() as MainActivityBackPressListener
            listener.onBackPressed()
        } else if (navController.currentDestination!!.id == R.id.noteListFragment) {
            super.moveTaskToBack(false)
        } else {
            navController.popBackStack()
        }
    }

    override fun backPreviousScreen() {
        navController.popBackStack()
    }

    lateinit var navController: NavController

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        navController = Navigation.findNavController(this, R.id.main_nav_host_fragment)
    }

}
